.github/workflows/ci.yml

```
@@ -0,0 +1,72 @@
+name: Elixir CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    name: Build and Test
+    runs-on: ubuntu-latest #kinda standard
+    services:
+      postgres:
+        image: postgres:17-alpine 
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust # Easiest for CI, no password needed
+          POSTGRES_DB: senpai_db_test 
+          POSTGRES_USER: postgres 
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Elixir
+        uses: erlef/setup-elixir@v1
+        with:
+          elixir-version: '1.18.3' # copied from the elixir -v command 🦜
```

```suggestion
          elixir-version: '1.18.3'
```

would be nice to automatically pull the versions from `.tool-versions` or the `flake.nix` but that might be hard?